rfcs/replace_ssh_with_ssm_for_linux_template/replace_ssh_with_ssm_for_linux_template.md

```diff
@@ -0,0 +1,238 @@
+7. Cloudformation stack create all resources declared in workspace template.
+8. User request to connect workspace, user will get the connections information including connection methods, such as "Https", "SSH" and "RDP(remote desktop protocol)".
+9. User access workspace resource through the information from above step.
+
+The step 1 to 5 are not related with connection method.
+The step 6, Service workbench will get output of cloudformation stack which contains the information of connection method, like below:
+For SSH connection:
+```
+  MetaConnection1Scheme:
+    Description: Protocol for connection
+    Value: ssh
+```
+For Https connection:
+```
+  MetaConnectionScheme:
+    Description: Protocol for connection 1
+    Value: https
+```    
+
+For RDP connection:
+```
+  MetaConnection1Scheme:
+    Description: Protocol for connection
+    Value: rdp
+```
+
+The step 8 and step 9 are the main logic that user connect workspace resources. Following section will show more detail.
+
+## Design of linux EC2 instance connection
+### UI experience of linux ec2 instance connection
+Let's know about current user experience of linux ec2 instance connection at first:
+![swb_connect_ec2_ssh_ui](./swb_connect_ec2_ssh_ui.png)
+
+1. User clicks "Connections" button, Web UI will show "Create Key" and "Use this SSH Key" if there is existing SSH Key.
+2. User choose a SSH Key, this SSH public key will be sent to EC2 instance, at the same time, UI will show the information to guide user how to access ec2 instance.
+3. User can set CIDRs to limit user to access EC2 instance.
+
+### Linux Workspace Connections Connection Sequence
+Current sequence of linux ec2 instance connection:
+![ssh-design.drawio](./ssh-design.drawio.png)
+
+1. User click "Connections" button, client side will send request to server side to get instanceId, connectionId and connection method. Web UI will render the page to match with connection method. In this case, the connection method is "ssh", Web UI will show the page of "ssh" to user according to "ssh" connection method.
+2. User choose a key which should be created by user before
+3. User click "connect", client side send request to backend to send public key to ec2 instance, then return all connection to client side, suce as "publicDnsName".
+4. Web UI will show the information to guide user how to access ec2 instance.
+5. User ssh to ec2 instance.
+
+# Design Summary
+The general approach proposed by this RFC is replacing ssh connection with AWS System Manager Session Manager. User can access ec2 instance through Session Manager, do not need to store and maintain ssh private key.
+
+## Web UI experience.
+Below is the experience after refactoring. It is similar with sagemaker workspace as below:
+![ssm-ui-experience](./ssm-ui-experience.png)
```

Comment:
I think the SSM login can be co-existence with SSH login. You might set an allowed regions list for EC2 instance connect feature. Disable the SSH login only when it's unavailable.

Reply:
Re-designed it to support both SSH and SSM in the regions where EC2 instance connect is available and support SSM for EC2 instance connect unavailable.